e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/HotNodePoolsAO.java

```diff
@@ -0,0 +1,230 @@
+import static com.codeborne.selenide.Selectors.byId;
+import static com.codeborne.selenide.Selectors.byText;
+import static com.codeborne.selenide.Selectors.byXpath;
+import static com.codeborne.selenide.Selenide.$;
+import static com.epam.pipeline.autotests.ao.Primitive.AUTOSCALED;
+import static com.epam.pipeline.autotests.ao.Primitive.CANCEL;
+import static com.epam.pipeline.autotests.ao.Primitive.CLOSE;
+import static com.epam.pipeline.autotests.ao.Primitive.CLOUD_REGION;
+import static com.epam.pipeline.autotests.ao.Primitive.CREATE;
+import static com.epam.pipeline.autotests.ao.Primitive.DELETE;
+import static com.epam.pipeline.autotests.ao.Primitive.DISK;
+import static com.epam.pipeline.autotests.ao.Primitive.EDIT;
+import static com.epam.pipeline.autotests.ao.Primitive.ENDS_ON;
+import static com.epam.pipeline.autotests.ao.Primitive.ENDS_ON_TIME;
+import static com.epam.pipeline.autotests.ao.Primitive.INSTANCE_TYPE;
+import static com.epam.pipeline.autotests.ao.Primitive.POOL_NAME;
+import static com.epam.pipeline.autotests.ao.Primitive.REFRESH;
+import static com.epam.pipeline.autotests.ao.Primitive.STARTS_ON;
+import static com.epam.pipeline.autotests.ao.Primitive.STARTS_ON_TIME;
+import static com.epam.pipeline.autotests.utils.PipelineSelectors.button;
+import static com.epam.pipeline.autotests.utils.Utils.nameWithoutGroup;
+import static java.lang.String.format;
+import static java.lang.String.valueOf;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.AssertJUnit.assertTrue;
+
+public class HotNodePoolsAO  implements AccessObject<ClusterMenuAO> {
+
+    private final Map<Primitive, SelenideElement> elements = initialiseElements(
+            entry(REFRESH, context().$(button("Refresh"))),
+            entry(CREATE, context().$(button("Create")))
+    );
+
+    public CreateHotNodePoolAO clickCreatePool() {
+        click(CREATE);
+        return new CreateHotNodePoolAO(this);
+    }
+
+    public NodeEntry searchForNodeEntry(String node) {
+        sleep(1, SECONDS);
+        SelenideElement entry = getNode(node).shouldBe(visible);
+        return new NodeEntry(this, node, entry);
+    }
+
+    private SelenideElement getNode(final String node) {
+        return $(byXpath(format(
+                ".//div[contains(@class, 'ool-card__header-container') and .//*[contains(text(), '%s')]]", node)))
+                .closest(".cp-hot-node-pool");
+    }
+
+    public HotNodePoolsAO waitUntilRunningNodesAppear(String poolName, int count) {
+        return waitUntilNodesAppear(poolName, 3, count);
+    }
+
+    public HotNodePoolsAO waitUntilActiveNodesAppear(String poolName, int count) {
+        return waitUntilNodesAppear(poolName, 1, count);
+    }
+
+    private HotNodePoolsAO waitUntilNodesAppear(String poolName, int nodeType, int count) {
+        NodeEntry node = searchForNodeEntry(poolName);
+        int attempt = 0;
+        int maxAttempts = 10;
```

Comment:
Variables is assigned but never accessed

Reply:
Condition using these variables has been added.